Bottom: 5a6828a5798f5606c540da479816a57c18c930a0
Top:    4a7ba383e003a6a7ab4ab4a62602f20b3eb90875
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-17 12:54:36 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index e6353c2..296b875 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -306,6 +306,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
